They handled the case when auth file is not present. But they did not handle the case when auth file is a bad file. For e.g. passing card file as an auth file.
And hence the program crashes with exit code 1.

They should have wrapped line 83 of Main.java with try/catch block.

